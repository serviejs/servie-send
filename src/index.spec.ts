import { send } from './index'
import { Request } from 'servie'
import { Readable } from 'stream'

describe('servie-send', () => {
  it('should send text', () => {
    const req = new Request({ url: '/' })
    const res = send(req, 'hello world')

    expect(res.status).toEqual(200)
    expect(res.headers).toMatchSnapshot()
    expect(res.body).toEqual('hello world')
  })

  it('should send an empty response', () => {
    const req = new Request({ url: '/' })
    const res = send(req, null)

    expect(res.status).toEqual(200)
    expect(res.headers).toMatchSnapshot()
    expect(res.body).toEqual('')
  })

  it('should send json', () => {
    const req = new Request({ url: '/' })
    const res = send(req, { hello: 'world' })

    expect(res.status).toEqual(200)
    expect(res.headers).toMatchSnapshot()
    expect(res.body).toEqual('{"hello":"world"}')
  })

  it('should send stream', () => {
    const req = new Request({ url: '/' })
    let chunk: string | null = 'hello world'

    const stream = new Readable({
      read () {
        this.push(chunk)
        chunk = null
      }
    })

    const res = send(req, stream)

    expect(res.status).toEqual(200)
    expect(res.headers).toMatchSnapshot()
  })

  it('should send 304 with matching etag', () => {
    const req = new Request({
      url: '/',
      headers: {
        'If-None-Match': '"0-47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU"'
      },
      body: ''
    })

    const res = send(req, '')

    expect(res.status).toEqual(304)
    expect(res.headers).toMatchSnapshot()
    expect(res.body).toEqual(undefined)
  })
})
