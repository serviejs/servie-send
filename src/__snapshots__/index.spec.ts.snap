// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`servie-send should send 304 with matching etag 1`] = `
Headers {
  "raw": Array [
    "ETag",
    "\\"0-47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU\\"",
  ],
}
`;

exports[`servie-send should send an empty response 1`] = `
Headers {
  "raw": Array [
    "Content-Type",
    "text/plain",
    "Content-Length",
    "0",
    "ETag",
    "\\"0-47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU\\"",
  ],
}
`;

exports[`servie-send should send json 1`] = `
Headers {
  "raw": Array [
    "Content-Type",
    "application/json",
    "Content-Length",
    "17",
    "ETag",
    "\\"h-k6I5cakU5erL8KjSUVTNownDwccvu5kU1Hxg88toFYg\\"",
  ],
}
`;

exports[`servie-send should send stream 1`] = `
Headers {
  "raw": Array [
    "Content-Type",
    "application/octet-stream",
  ],
}
`;

exports[`servie-send should send text 1`] = `
Headers {
  "raw": Array [
    "Content-Type",
    "text/plain",
    "Content-Length",
    "11",
    "ETag",
    "\\"b-uU0nuZNNPgilLlLX2n2r+sSE7+N6U4DukIj3rOLvzek\\"",
  ],
}
`;
